src/main/java/org/opensearch/geospatial/ip2geo/common/GeoIpDataHelper.java

```diff
@@ -0,0 +1,355 @@
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.common;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.opensearch.OpenSearchException;
+import org.opensearch.SpecialPermission;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.bulk.BulkRequest;
+import org.opensearch.action.bulk.BulkResponse;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.search.MultiSearchRequestBuilder;
+import org.opensearch.action.search.MultiSearchResponse;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.client.Client;
+import org.opensearch.client.Requests;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.SuppressForbidden;
+import org.opensearch.common.collect.Tuple;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.index.query.QueryBuilders;
+
+/**
+ * Helper class for GeoIp data
+ */
+@Log4j2
+public class GeoIpDataHelper {
+    private static final String IP_RANGE_FIELD_NAME = "_cidr";
+    private static final String DATA_FIELD_NAME = "_data";
+    private static final Tuple<String, Integer> INDEX_SETTING_NUM_OF_SHARDS = new Tuple<>("index.number_of_shards", 1);
+    private static final Tuple<String, String> INDEX_SETTING_AUTO_EXPAND_REPLICAS = new Tuple<>("index.auto_expand_replicas", "0-all");
+
+    /**
+     * Create an index of single shard with auto expand replicas to all nodes
+     *
+     * @param clusterService cluster service
+     * @param client client
+     * @param indexName index name
+     * @param timeout timeout
+     */
+    public static void createIndexIfNotExists(
+        final ClusterService clusterService,
+        final Client client,
+        final String indexName,
+        final TimeValue timeout
+    ) {
+        if (clusterService.state().metadata().hasIndex(indexName) == true) {
+            log.info("Index {} already exist", indexName);
```

Comment:
nit: Do we need this info here? I believe, adding log when you are creating index for first time is enough.

Reply:
Removed